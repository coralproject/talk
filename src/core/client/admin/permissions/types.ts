import { GQLUSER_ROLE_RL } from "coral-framework/schema";

export type PermissionMap<AbilityType extends string, ContextType> = {
  [abilityType in AbilityType]: {
    [role in GQLUSER_ROLE_RL]?: (context?: ContextType) => boolean;
  };
};
