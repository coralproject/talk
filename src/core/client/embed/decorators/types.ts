import pym from "pym.js";

export type CleanupCallback = () => void;
export type Decorator = (pym: pym.Parent) => CleanupCallback | void;
