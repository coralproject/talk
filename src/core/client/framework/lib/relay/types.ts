import { _RefType } from "react-relay";

export type FragmentKeys<T> = {
  [P in keyof T]: T[P] extends _RefType<any> | null ? P : never
}[keyof T];

export type FragmentKeysNoLocal<T> = Exclude<FragmentKeys<T>, "local">;
