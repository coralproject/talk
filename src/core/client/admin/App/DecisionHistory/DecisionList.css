.root {
  margin: 0;
  padding: 0;
  list-style-type: none;

  & > * {
    border-bottom: 1px solid var(--v2-palette-decision-history-border);
  }
  & > *:last-child {
    border-bottom: 0;
  }
}
