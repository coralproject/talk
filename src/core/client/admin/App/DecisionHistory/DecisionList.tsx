import React, { FunctionComponent } from "react";

import styles from "./DecisionList.css";

interface DecisionListProps {
  children?: React.ReactNode;
}

const DecisionList: FunctionComponent<DecisionListProps> = (props) => (
  <ul className={styles.root}>{props.children}</ul>
);

export default DecisionList;
