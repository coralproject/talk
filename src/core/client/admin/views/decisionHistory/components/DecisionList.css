.root {
  margin: 0;
  padding: 0;
  list-style-type: none;

  & > * {
    border-bottom: 1px solid var(--palette-divider);
  }
  & > *:last-child {
    border-bottom: 0;
  }
}
