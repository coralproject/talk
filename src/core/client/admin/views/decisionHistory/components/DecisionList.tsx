import React, { StatelessComponent } from "react";

import styles from "./DecisionList.css";

const DecisionList: StatelessComponent = props => (
  <ul className={styles.root}>{props.children}</ul>
);

export default DecisionList;
