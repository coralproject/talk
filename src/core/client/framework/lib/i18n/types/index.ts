export { default as FluentShortNumber } from "./FluentShortNumber";
