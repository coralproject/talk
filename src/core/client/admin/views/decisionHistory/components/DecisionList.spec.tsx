import { shallow } from "enzyme";
import React from "react";

import { PropTypesOf } from "talk-framework/types";

import DecisionList from "./DecisionList";

it("renders correctly", () => {
  const props: PropTypesOf<typeof DecisionList> = {
    children: "children",
  };
  const wrapper = shallow(<DecisionList {...props} />);
  expect(wrapper).toMatchSnapshot();
});
