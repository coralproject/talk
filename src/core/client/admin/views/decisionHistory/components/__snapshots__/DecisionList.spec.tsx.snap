// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="DecisionList-root"
>
  children
</ul>
`;
